lfs/config.go

```diff
@@ -105,41 +116,99 @@ func (c *Configuration) RemoteEndpoint(remote string) Endpoint {
 	if len(remote) == 0 {
 		remote = defaultRemote
 	}
-
 	if url, ok := c.GitConfig("remote." + remote + ".lfsurl"); ok {
-		return Endpoint{Url: url}
+		return NewEndpoint(url)
 	}
 
 	if url, ok := c.GitConfig("remote." + remote + ".url"); ok {
-		return remoteEndpointFromUrl(url)
+		return NewEndpointFromCloneURL(url)
 	}
 
 	return Endpoint{}
 }
 
-func remoteEndpointFromUrl(url string) Endpoint {
-	e := Endpoint{Url: url}
-
-	if !httpPrefixRe.MatchString(url) {
-		pieces := strings.SplitN(url, ":", 2)
-		hostPieces := strings.SplitN(pieces[0], "@", 2)
-		if len(hostPieces) < 2 {
-			e.Url = "<unknown>"
-			return e
+const ENDPOINT_URL_UNKNOWN = "<unknown>"
+
+// Create a new endpoint from a URL associated with a git clone URL
+// The difference to NewEndpoint is that it appends [.git]/info/lfs to the URL since it
+// is the clone URL
+func NewEndpointFromCloneURL(url string) Endpoint {
+	e := NewEndpoint(url)
+	if e.Url != ENDPOINT_URL_UNKNOWN {
+		// When using main remote URL for HTTP, append info/lfs
+		if path.Ext(url) == ".git" {
+			e.Url += "/info/lfs"
+		} else {
+			e.Url += ".git/info/lfs"
 		}
-
-		e.SshUserAndHost = pieces[0]
-		e.SshPath = pieces[1]
-		e.Url = fmt.Sprintf("https://%s/%s", hostPieces[1], pieces[1])
 	}
+	return e
+}
+
+// Create a new endpoint from a general URL
+func NewEndpoint(urlstr string) Endpoint {
+	endpoint := Endpoint{Url: urlstr}
+	// Check for SSH URLs
+	// Support ssh://user@host.com/path/to/repo and user@host.com:path/to/repo
+	u, err := url.Parse(urlstr)
+	if err == nil {
```

Comment:
OK - rebased and will refactor a little in next commit. 

You did VB/Access too eh? We should commiserate over a beer one day ;)